group-coordinator/src/test/java/org/apache/kafka/coordinator/group/GroupMetadataManagerTest.java

```diff
@@ -172,19 +186,21 @@ public List<Record> build(TopicsImage topicsImage) {
             });
 
             // Add subscription metadata.
-            Map<String, TopicMetadata> subscriptionMetadata = new HashMap<>();
-            members.forEach((memberId, member) -> {
-                member.subscribedTopicNames().forEach(topicName -> {
-                    TopicImage topicImage = topicsImage.getTopic(topicName);
-                    if (topicImage != null) {
-                        subscriptionMetadata.put(topicName, new TopicMetadata(
-                            topicImage.id(),
-                            topicImage.name(),
-                            topicImage.partitions().size()
-                        ));
-                    }
+            if (subscriptionMetadata == null) {
```

Comment:
If subscriptionMetadata is not null, that means we just want to take what we are already given and not generate it based on the subscribedNames and the image?
Is this done to optimize or are there cases where we want to have something different than what we would have generated?

Reply:
Most of tests are just fine with the auto-generated subscription metadata. However, the new ones need specific subscription metadata to verify the check. This is why I extended this builder to either accept the subscription metadata to use or to auto-generate it.